toolbox/fdc3-workbench/src/components/Intents.tsx

```
@@ -166,6 +218,66 @@ export const Intents = observer(() => {
 		}
 	};
 
+	useEffect(() => {
+		const fetchIntents = async () => {
+			try {
+				if(raiseIntentContext) {
+					let appIntents = await fdc3.findIntentsByContext(toJS(raiseIntentContext));
+					if (appIntents) {
+						setIntentObjects(appIntents);
+						setIntentsForContext(appIntents.map(({intent}: {intent:any})=>{
+							return {
+								title: intent.displayName,
+								value: intent.displayName
+							}
+						}));
+					}
+				}
+			} catch (e){
+				setRaiseIntentError(" no intents found");
+			}
+		}
+		fetchIntents().catch(()=>{
+			setRaiseIntentError(" no intents found");
+		});
+		
+	}, [raiseIntentContext]);
+
+	useEffect(() => {
+		if(intentObjects){
+			let targets = intentObjects.find((obj) => obj.intent.displayName === intentValue?.value);
+			if(targets?.apps){
+				setIntentTargets(targets.apps);
+			}
+		}
+	}, [intentValue]);
+
+	useEffect(() => {
+		const fetchIntents = async () => {
+			try {
+				if(raiseIntentWithContextContext) {
+					let appIntentsForContext = await fdc3.findIntentsByContext(toJS(raiseIntentWithContextContext));
+					if (appIntentsForContext) {
+						let pairObject: any[] = [];
+						appIntentsForContext.forEach((intent)=>{
+							intent?.apps.forEach((app)=>{
+								pairObject.push( {
+									name: `${app.name} - ${intent.intent.name}`,
+									app
+								});
+							})
+						});
+						setContextIntentObjects(pairObject as any[]);
+					}
+				}
+			} catch (e){
```

probably just need 1 catch but do we need to raise an error here?